### PR TITLE
Problem: withdraw delegation reward is not working with ledger (fix #…

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution:
change MsgWithdrawDelegatorReward type 
from 'cosmos-sdk/MsgWithdrawDelegationReward'
to 'cosmos-sdk/MsgWithdrawDelegationReward'

### CODE REVIEW

**Review by crypto-matto — APPROVED** [review]

**Review by eddycpc — APPROVED** [review]

**Review by linfeng-crypto — APPROVED** [review]

**Comment by calvinaco** [comment]
Will this affect our protobuf version of the message? Because I see it is only changed in Amino JSON version.

File: lib/src/transaction/msg/distribution/MsgWithdrawDelegatorReward.spec.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ describe('Testing MsgWithdrawDelegatorReward', function () {
             });
 
             const rawMsg: legacyAmino.Msg = {
-                type: 'cosmos-sdk/MsgWithdrawDelegatorReward',
```
Comment (eddycpc) [inline lib/src/transaction/msg/distribution/MsgWithdrawDelegatorReward.spec.ts:80]: Thanks, @calvinaco for noticing this.

This is definitely a breaking change for Protobuf delegator reward withdrawal which was working already before and didn't need to be changed here.

**Review by eddycpc — CHANGES_REQUESTED** [review]

File: lib/src/transaction/msg/distribution/MsgWithdrawDelegatorReward.spec.ts (hunk 1 above)
Comment (calvinaco) [reply lib/src/transaction/msg/distribution/MsgWithdrawDelegatorReward.spec.ts:80]: cc @leejw51crypto

Comment (leejw51crypto) [reply lib/src/transaction/msg/distribution/MsgWithdrawDelegatorReward.spec.ts:80]: i'll check

Comment (leejw51crypto) [reply lib/src/transaction/msg/distribution/MsgWithdrawDelegatorReward.spec.ts:80]: export const COSMOS_MSG_TYPEURL = {
    MsgBeginRedelegate: '/cosmos.staking.v1beta1.MsgBeginRedelegate',
    Coin: '/cosmos.base.v1beta1.Coin',
    MsgSend: '/cosmos.bank.v1beta1.MsgSend',
    MsgWithdrawDelegatorReward: '/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward',
    MsgCreateValidator: '/cosmos.staking.v1beta1.MsgCreateValidator',
    MsgEditValidator: '/cosmos.staking.v1beta1.MsgEditValidator',
    MsgDelegate: '/cosmos.staking.v1beta1.MsgDelegate',
    MsgUndelegate: '/cosmos.staking.v1beta1.MsgUndelegate',
    MsgWithdrawValidatorCommission: '/cosmos.distribution.v1beta1.MsgWithdrawValidatorCommission',
    MsgDeposit: '/cosmos.gov.v1beta1.MsgDeposit',
    MsgVote: '/cosmos.gov.v1beta1.MsgVote',
    MsgSubmitProposal: '/cosmos.gov.v1beta1.MsgSubmitProposal',
    PubKey: {
        ed25519: '/cosmos.crypto.ed25519.PubKey',
    },
};



        // eslint-disable-next-line class-methods-use-this
        toRawAminoMsg(): legacyAmino.Msg {
            return {
                type: 'cosmos-sdk/MsgWithdrawDelegationReward',
                value: {
                    delegator_address: this.delegatorAddress,
                    validator_address: this.validatorAddress,
                },
            } as legacyAmino.MsgWithdrawDelegatorReward;
        }

        /**
         * Returns the raw Msg representation of MsgWithdrawDelegatorReward
         * @returns {Msg}
         */
        toRawMsg(): Msg {
            return {
                typeUrl: COSMOS_MSG_TYPEURL.MsgWithdrawDelegatorReward,
                value: {
                    delegatorAddress: this.delegatorAddress,
                    validatorAddress: this.validatorAddress,
                },
            };
        }

Comment (leejw51crypto) [reply lib/src/transaction/msg/distribution/MsgWithdrawDelegatorReward.spec.ts:80]: protobuf used toRawMsg, it is still using '/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward',

Comment (leejw51crypto) [reply lib/src/transaction/msg/distribution/MsgWithdrawDelegatorReward.spec.ts:80]: image

protobuf
withdraw delegator reward, working fine

**Review by calvinaco — APPROVED** [review]

**Review by eddycpc — APPROVED** [review]

**Pull request merged**